src/backend/InvenTree/plugin/builtin/labels/label_sheet.py

```diff
@@ -64,9 +65,22 @@ class InvenTreeLabelSheetPlugin(LabelPrintingMixin, SettingsMixin, InvenTreePlug
 
     BLOCKING_PRINT = True
 
-    SETTINGS = {}
+    SETTINGS = {
+        'DEBUG': {
+            'name': _('Debug mode'),
+            'description': _('Enable debug mode - returns raw HTML instead of PDF'),
+            'validator': bool,
+            'default': False,
+        }
+    }
 
     PrintingOptionsSerializer = LabelPrintingOptionsSerializer
+    debug = None
+
+    def in_debug_mode(self):
+        """Check if the plugin is printing in debug mode."""
+        if self.debug is None:
+            self.debug = str2bool(self.get_setting('DEBUG'))
```

Comment:
Oops, this function should return a value, right?

Reply:
Also, what's the purpose of setting self.debug if it's not defined? The plugin settings are cached. And secondly this means that we cannot disable debug mode at runtime anymore.